Add preprocess to ci

lucylq opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #6553
* __->__ #6544

1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

lucylq committed:
Add preprocess to ci

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and then re-using them on different inputs.

Test Plan:
```
python -m unittest examples.models.llama3_2_vision.preprocess.test_preprocess
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and then re-using them on different inputs.

Test Plan:
```
python -m unittest examples.models.llama3_2_vision.preprocess.test_preprocess
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
python -m unittest examples.models.llama3_2_vision.preprocess.test_preprocess
```

[ghstack-poisoned]

larryliu0820 commented on .github/workflows/pull.yml:
```diff
@@ -231,6 +231,34 @@ jobs:
+        bash examples/models/llama3_2_vision/install_requirements.sh
+
+        # run python unittest
+        python -m unittest examples.models.llama3_2_vision.preprocess.test_preprocess
```
Hmm I don't think we need a new job right? Just add it to `pytest.ini` should be good enough.

lucylq replied:
updated to use pytest

larryliu0820 replied:
So not sure if you need to run `bash examples/models/llama3_2_vision/install_requirements.sh` though. Let's wait and see.

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
python -m unittest examples.models.llama3_2_vision.preprocess.test_preprocess
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]

lucylq committed:
Update on "Add preprocess to ci"


1. Run `test_preprocess.py` in ci.
2. Refactor `test_preprocess.py` by creating the model artifacts (eager, exported, et) once and re-use them on different inputs (previously, create model artifacts for each test).

Test Plan:
```
pytest -c /dev/null -v -n auto examples/models/llama3_2_vision/preprocess/
```

[ghstack-poisoned]
